circuits/ts/__tests__/ProcessMessages.test.ts

```diff
@@ -117,10 +114,21 @@ describe("ProcessMessage circuit", function () {
       commands.push(command2);
       poll.publishMessage(message2, ecdhKeypair2.pubKey);
 
-      poll.messageAq.mergeSubRoots(0);
-      poll.messageAq.merge(treeDepths.messageTreeDepth);
-
-      expect(poll.messageTree.root.toString()).to.be.eq(poll.messageAq.getRoot(treeDepths.messageTreeDepth).toString());
+      // Use the accumulator queue to compare the root of the message tree
+      const accumulatorQueue: AccQueue = new AccQueue(
+        treeDepths.messageTreeSubDepth,
+        STATE_TREE_ARITY,
+        NOTHING_UP_MY_SLEEVE,
+      );
+      accumulatorQueue.enqueue(message.hash(ecdhKeypair.pubKey));
+      accumulatorQueue.enqueue(message2.hash(ecdhKeypair2.pubKey));
+      accumulatorQueue.mergeSubRoots(0);
+      accumulatorQueue.merge(treeDepths.messageTreeDepth);
+
+      // should this here?
```

Comment:
I guess it doesn't hurt to do this extra check, just to ensure we start the test suite with the correct state. Up to you what to do there, could even remove it.

Been also thinking that we could have an utility function do this, as we are doing the same process throughout different test cases

Reply:
Yeah, I thought it would be more appropriate to validate the integrity of the Merkle tree in maci-core or maci-contracts. It seems like there is some redundancy, but I need to take a closer look to organize it. Anyway, it was pushed accidentally; it was meant to be wip note, but thanks for sharing your thoughts!